### ISSUE TITLE
Reverse lookup in the envelope is never initialized

### ISSUE THREAD

**Issue by NicolasLM** [slimta/python-slimta#117]
The documentation of `Envelope` mentions that `self.client['host']` contains the reverse lookup of the client IP, however it is always set to `None`.

I tracked down the issue a bit and it seems that the result of the Greenlet thread `PtrLookup` is never used anywhere.

Congratulation on writing this library, the code is well written and easily extensible :)

**Comment by NicolasLM** [comment]
Actually this issue has unexpected consequences. As a reverse is not available in the Envelope, the `Received` header added by slimta does not contain a reverse:

```
Received: from my-laptop (unknown [127.0.0.1])
```

Thus Spamassassin triggers the rule `RDNS_NONE` for all emails received.

I took the liberty to fix the issue in #119, it now adds the correct header:

```
Received: from my-laptop (localhost [127.0.0.1])
```

**Comment by icgood** [comment]
Great catch, sorry it took me a little while to merge, but your fix looks great and thanks for adding the test.

**Comment by icgood** [comment]
By the way, I have tagged this fix in version [4.0.1](https://github.com/slimta/python-slimta/releases/tag/4.0.1) and uploaded to PyPI.
